modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -69,6 +71,68 @@ def wait(*args, **kwargs):
     return func
 
 
+def progress_bar_wrapper(f):
+    """
+    Wrap computation function inside a progress bar.
+
+    Spawns another thread which displays a progress bar showing
+    estimated completion time.
+
+    Parameters
+    ----------
+    f : callable
+        The name of the function to be wrapped.
+
+    Returns
+    -------
+    callable
+        Decorated version of `f` which reports progress.
+    """
+    if Engine.get() != "Ray":
+        if ProgressBar.get():
+            raise NotImplementedError(
+                f"ProgressBar doesn't work with '{Engine.get()}' engine."
+            )
+        else:
+            return f
```

Comment:
These lines - the main changes that allow us to move progress bar usage into base class.